jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsResourceCache.kt

```diff
@@ -296,7 +298,9 @@ class DefaultAwsResourceCache(
                 val result = cache.compute(context.cacheKey) { _, value ->
                     fetchIfNeeded(context, value as Entry<T>?)
                 } as Entry<T>
-                context.future.complete(result.value)
+                launch {
+                    context.future.complete(result.value.await())
```

Comment:
if this throws it won't be caught by the catch block anymore?

Reply:
fixed